vm/src/circuit/r1cs.rs

```diff
@@ -324,9 +324,21 @@ impl R1CS {
     }
 }
 
+fn unique_values<Elm: Eq>(set: &[Elm]) -> bool {
```

Comment:
I know this code will only be invoked on debug builds so performance doesn't matter all that much, but you can do this in `O(n)` rather than `O(n^2)` by using a `HashSet`.

Reply:
will do that